kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java

```diff
@@ -384,4 +365,32 @@ public Subscription create(Subscription subscription) {
     }
 
 
+    @Override
+    public final void createSubscriptionPacks() {
+        if (subscriptionPackDataService.byName("childPack") == null) {
+            createSubscriptionPack("childPack", SubscriptionPackType.CHILD, CHILD_PACK_WEEKS, 1);
+        }
+        if (subscriptionPackDataService.byName("pregnancyPack") == null) {
+            createSubscriptionPack("pregnancyPack", SubscriptionPackType.PREGNANCY, PREGNANCY_PACK_WEEKS, 2);
+        }
+    }
+
+
+    private void createSubscriptionPack(String name, SubscriptionPackType type, int weeks,
```

Comment:
Yep, I think it's fine to duplicate some IT code (especially simple data
population)
But I really don't feel strongly about it.


On Tue, May 26, 2015 at 9:02 PM, Lauren Lavoie <notifications@github.com>
wrote:

> In
> kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java
> <https://github.com/motech-implementations/mim/pull/280#discussion_r31101762>
> :
>
> > @@ -384,4 +365,32 @@ public Subscription create(Subscription subscription) {
> >      }
> >
> >
> > +    @Override
> > +    public final void createSubscriptionPacks() {
> > +        if (subscriptionPackDataService.byName("childPack") == null) {
> > +            createSubscriptionPack("childPack", SubscriptionPackType.CHILD, CHILD_PACK_WEEKS, 1);
> > +        }
> > +        if (subscriptionPackDataService.byName("pregnancyPack") == null) {
> > +            createSubscriptionPack("pregnancyPack", SubscriptionPackType.PREGNANCY, PREGNANCY_PACK_WEEKS, 2);
> > +        }
> > +    }
> > +
> > +
> > +    private void createSubscriptionPack(String name, SubscriptionPackType type, int weeks,
>
> But then would we have to duplicate them in both the API IT code and the
> Kilkari IT code? Both modules use them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/pull/280/files#r31101762>.
>



-- 
Frank Huster
Mobile Health Innovations
Grameen Foundation
+1.206.325.6690 x201